geoh5py/workspace/workspace.py

```diff
@@ -626,9 +626,9 @@ def finalize(self, file: str | h5py.File | None = None):
         :param file: :obj:`h5py.File` or name of the target geoh5 file
         """
         for entity in (
-            cast(List["Entity"], self.objects)
-            + cast(List["Entity"], self.groups)
-            + cast(List["Entity"], self.data)
+            cast(list[Entity], self.objects)  # pylint: disable=unsubscriptable-object
```

Comment:
although cast is ignored at runtime, `pylint` does not like the new type hint syntax there